include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -159,51 +160,62 @@ class pairwise_alignment_algorithm : protected policies_t...
      * \param[in] sequence2 The second sequence to compute the alignment for.
      */
     template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
-    int32_t compute_matrix(sequence1_t & sequence1, sequence2_t & sequence2)
+    int32_t compute_matrix(sequence1_t && sequence1, sequence2_t && sequence2)
     {
         // ---------------------------------------------------------------------
         // Initialisation phase: allocate memory and initialise first column.
         // ---------------------------------------------------------------------
 
         this->reset_optimum(); // Reset the tracker for the new alignment computation.
 
-        thread_local score_matrix_single_column<int32_t> local_matrix{};
-        local_matrix.reset_matrix(std::forward<sequence1_t>(sequence1), std::forward<sequence2_t>(sequence2));
-        auto matrix_iter = local_matrix.begin();
+        thread_local score_matrix_single_column<int32_t> local_score_matrix{};
+        coordinate_matrix local_index_matrix{};
 
-        initialise_column(*matrix_iter, sequence2);
+        size_t number_of_columns = std::ranges::distance(sequence1) + 1;
+        size_t number_of_rows = std::ranges::distance(sequence2) + 1;
+
+        local_score_matrix.resize(column_index_type{number_of_columns}, row_index_type{number_of_rows});
+        local_index_matrix.resize(column_index_type{number_of_columns}, row_index_type{number_of_rows});
+
+        auto alignment_matrix_it = local_score_matrix.begin();
```

Comment:
```suggestion
        auto score_matrix_it = local_score_matrix.begin();
```
To be consistent with the naming of `indexed_matrix_it` for `local_index_matrix`?

Reply:
In general I agree, but usually the alignment matrix is more than just the score matrix. We only have the score matrix right now. But later we want to have an alignment matrix which combines score and trace matrix. Then I only have to adapt the one line and do. Not forget to rename the other variables.